pkg/controller/controllerutil/controllerutil.go

```diff
@@ -32,6 +34,33 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
 )
 
+const inClusterNamespacePath = "/var/run/secrets/kubernetes.io/serviceaccount/namespace"
+
+var inClusterNamespace string
+
+// GetInClusterNamespace returns the namespace by which the controller is running under.
+func GetInClusterNamespace() (string, error) {
+	if inClusterNamespace != "" {
+		return inClusterNamespace, nil
+	}
+	// Check whether the namespace file exists.
+	// If not, we are not running in cluster so can't guess the namespace.
+	_, err := os.Stat(inClusterNamespacePath)
+	if os.IsNotExist(err) {
+		return "", fmt.Errorf("not running in-cluster")
+	} else if err != nil {
+		return "", fmt.Errorf("error checking namespace file: %w", err)
```

Comment:
Do we actually need this part? Couldn't wew check the result of `ReadFile` instead?

Reply:
Going to check. I do remember doing this before somewhere else for some obscure POSIX error reason.

Reply:
Wasn't necessary to do Stat separately, have cleaned up.